src/main/java/org/fundacionjala/coding/melvi/BankOcr.java

```diff
@@ -0,0 +1,263 @@
+ */
+public class BankOcr {
+    private static final Logger LOGGER = Logger.getLogger(BankOcr.class.getName());
+    static final String PATTERN0 = " _ | ||_|";
+    static final String PATTERN1 = "     |  |";
+    static final String PATTERN2 = " _  _||_ ";
+    static final String PATTERN3 = " _  _| _|";
+    static final String PATTERN4 = "   |_|  |";
+    static final String PATTERN5 = " _ |_  _|";
+    static final String PATTERN6 = " _ |_ |_|";
+    static final String PATTERN7 = " _   |  |";
+    static final String PATTERN8 = " _ |_||_|";
+    static final String PATTERN9 = " _ |_| _|";
+    public static final int TRHEE = 3;
+    public static final int INT = 4;
+    public static final int NINE = 9;
+    public static final int ONCE = 11;
+
+    private Map<String, String> charMap = new HashMap<>();
+
+    /**
+     * Class Constructor.
+     */
+    public BankOcr() {
+        charMap.put(PATTERN0, "0");
+        charMap.put(PATTERN1, "1");
+        charMap.put(PATTERN2, "2");
+        charMap.put(PATTERN3, "3");
+        charMap.put(PATTERN4, "4");
+        charMap.put(PATTERN5, "5");
+        charMap.put(PATTERN6, "6");
+        charMap.put(PATTERN7, "7");
+        charMap.put(PATTERN8, "8");
+        charMap.put(PATTERN9, "9");
+    }
+
+    /**
+     * @param chars array to initialize
+     * @return initialized array
+     */
+    private StringBuilder[] initializeChars(final StringBuilder[] chars) {
+        for (int i = 0; i < chars.length; i++) {
+            chars[i] = new StringBuilder();
+        }
+        return chars;
+    }
+
+    /**
+     * @param chars array to clean
+     * @return clean array
+     */
+    private StringBuilder[] cleanChars(final StringBuilder[] chars) {
+        for (int i = 0; i < chars.length; i++) {
+            chars[i].setLength(0);
+        }
+        return chars;
+    }
+
+    /**
+     * @param chars chars array
+     * @param line  line to process
+     * @return array with extracted chars
+     */
+    private StringBuilder[] extractCharsFromLine(final StringBuilder[] chars, final String line) {
+        for (int c = 0; c < chars.length; c++) {
+            chars[c].append(line.substring(c * TRHEE, c * TRHEE + TRHEE));
+        }
+        return chars;
+    }
+
+    /**
+     * @param chars array to process
+     * @return number extracted
+     */
+    private String identifyChars(final StringBuilder[] chars) {
+        String number = "";
+        for (int i = 0; i < chars.length; i++) {
+            number = number.concat(identifyChar(chars[i].toString()));
+        }
+
+        return number;
+    }
+
+
+    /**
+     * @param filePath url of file finePath
+     * @return arrayList<String> with the list of numbers
+     */
+    public List<String> scanOcr(final String filePath) {
+        ArrayList<String> resultado = new ArrayList<>();
+        StringBuilder[] chars = new StringBuilder[BankOcr.NINE];
+        chars = initializeChars(chars);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment; "chars" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4165)